src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/GenericSecurityToken.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Net;
+using System.Security.Principal;
+using CoreWCF.IdentityModel;
+using CoreWCF.IdentityModel.Tokens;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    public class GenericSecurityToken : SecurityToken
+    {
+        private string _id;
+        private DateTime _effectiveTime;
+        private DateTime _expirationTime;
+
+        internal GenericSecurityToken(string servicePrincipalName, TokenImpersonationLevel tokenImpersonationLevel, NetworkCredential networkCredential, string id)
```

Comment:
Change servicePrincipalName to just name as SPN is Kerberos specific. I don't think you need the impersonation level either.